src/components/channels/ChannelsSubscriber.js

```diff
@@ -23,99 +23,117 @@ const VIEWER_PALETTE = [
 
 const STANDARD_MAX = 65535;
 
-export default class ChannelsSubscriber extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { colorValues: {}, channelsOn: {}, rangeValues: {} };
-    this.setSliderValue = this.setSliderValue.bind(this);
-    this.setColorValue = this.setColorValue.bind(this);
-    this.toggleChannel = this.toggleChannel.bind(this);
-    this.componentWillUnmount = this.componentWillUnmount.bind(this);
-  }
+export default function ChannelsSubscriber({ onReady, removeGridComponent }) {
+  const [channelNames, setChannelNames] = useState(null);
+  const [colorValues, setColorValues] = useState(null);
+  const [sliderValues, setSliderValues] = useState(null);
+  const [channelIsOn, setChannelIsOn] = useState(null);
+  const [sliderDomains, setSliderDomains] = useState(null);
+  const [eventId, setEventId] = useState(null);
 
-  // eslint-disable-next-line camelcase
-  UNSAFE_componentWillMount() {
-    this.rasterAddToken = PubSub.subscribe(RASTER_ADD, this.rasterAddSubscriber.bind(this));
-  }
+  const memoizedOnReady = useCallback(onReady, []);
 
-  componentDidMount() {
-    const { onReady } = this.props;
-    onReady();
-  }
+  useEffect(() => {
+    function handleRasterAdd(msg, raster) {
+      const { domains, id, channelNames: cNames } = raster;
 
-  componentWillUnmount() {
-    PubSub.unsubscribe(this.rasterAddToken);
-  }
+      const initialColorValues = domains.map((_, i) => VIEWER_PALETTE[i]);
```

Comment:
It might make sense to allow colors to be defined, but this is something I would think should live in the app configuration rather than including this in the schema. We need to strike some balance of guessing a good configuration while not requiring the schema to be super complex. I would rather save a "selection" and "channel colors" state config in the future.

Reply:
Maybe a regex for known things like `DAPI` or the like?  I can open an issue here if you think something like that would work.

Reply:
Again, I'm not sure if this is something we want bake into vitessce in this way. I want to avoid custom, hardcoded logic. I would prefer a state configuration which we are working towards.

Reply:
Maybe I’m confused - what do you mean by “State Configuration?”

Reply:
Data `sources` are the entry point to tell vitessce _what_ to put on the screen. However, if you refresh the page all "state" is lost (i.e. what cells are selected, what colors are used, what slider values are, what zoom level youre in, etc). Ultimately, we also want to allow the user to save the "state" of their application (sliderValues, colorValues, zoom, genes, etc) so that they can share a url with someone else and they see exactly what you see on your screen. 

What I am suggesting here is that colors should not live in the data source. Instead, a user should supply a `source` and then configure and save/export the state that they want (sliderValues, colors, and which channels are shown by default).

Reply:
Ah, ok, that makes sense.